From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Sat, 25 Jun 2022 17:05:42 +0200
Subject: [PATCH 1/1] [WasmFS] mmap: avoid setting *allocated too early

Setting `*allocated` here is probably too early, as read errors
may occur (causing the pointer to be `free()`'d).

Upstream-Status: Accepted [https://github.com/emscripten-core/emscripten/commit/77da62969eb0b5efe4ce4a0c9ce67c45b9e32638]

diff --git a/system/lib/wasmfs/syscalls.cpp b/system/lib/wasmfs/syscalls.cpp
index 1111111..2222222 100644
--- a/system/lib/wasmfs/syscalls.cpp
+++ b/system/lib/wasmfs/syscalls.cpp
@@ -1529,7 +1529,6 @@ intptr_t _mmap_js(
   if (!ptr) {
     return -ENOMEM;
   }
-  *allocated = true;
 
   auto written = file->locked().read(ptr, length, offset);
   if (written < 0) {
